app/middleware/getServices.js

```diff
@@ -52,22 +58,25 @@ function getServices(req, res, next) {
   const endTimer = esGetServiceHistogram.startTimer();
   const timerLabel = {};
   timerLabel[esQueryLabelName] = esQuery.label;
-  esClient
-    .client
-    .search(esQuery.query)
-    .then((results) => {
+
+  try {
+    const results = await esClient.client.search(esQuery.query);
+
+    const logResults = () => {
       endTimer(timerLabel);
       log.info({
         esQuery,
         location,
         postcodeLocationDetails,
         resultCount: results.hits.total,
       }, 'getServices ES query and params');
-      res.locals.resultsCount = results.hits.total;
-      mapResults(results, res);
-    })
-    .then(next)
-    .catch(error => handleError(error, next));
+    };
+
+    processResults(results, res, logResults);
```

Comment:
is there much benefit in defining the local function `logResults`, passing it to `processResults` then immediately calling `logResults`,  over executing the 'endTimer' and 'log.info' before calling `processResults`?

Reply:
Probably not with what I'de done here. I meant to move it outside of the try catch, which I've done in the latest change. The reason for doing it is to try and make it a bit clearer what is happening within the try catch.